packages/components/src/draggable/stories/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Draggable from '../';
+import Dashicon from '../../dashicon';
+
+export default { title: 'Draggable', component: Draggable };
+
+const Box = ( props ) => {
+	return (
+		<div
+			{ ...props }
+			style={ {
+				alignItems: 'center',
+				display: 'flex',
+				justifyContent: 'center',
+				width: 100,
+				height: 100,
+				background: '#ddd',
+			} }
+		/>
+	);
+};
+
+const DraggalbeExample = () => {
+	const [ isDragging, setDragging ] = useState( false );
+
+	// Allow for the use of ID in the example
+	/* eslint-disable no-restricted-syntax */
+	return (
+		<div>
+			<p>Is Dragging? { isDragging ? 'Yes' : 'No' }</p>
+			<div id="draggable-example-box" style={ { display: 'inline-flex' } }>
```

Comment:
I'm a bit concerned about the implementation of the `Drabbable` component. Why do you need to wrap it with a `div` which has the `id` which is referenced with `elementId` prop?

It's completely unrelated to this PR, but I'd be in favor of opening a follow-up issue which makes it easier to use.

Taking the example from README:

```jsx
import { Dashicon, Draggable, Panel, PanelBody } from '@wordpress/components';

const MyDraggable = ( { onDragStart, onDragEnd } ) => (
	<div id="draggable-panel">
		<Panel header="Draggable panel" >
			<PanelBody>
				<Draggable
					elementId="draggable-panel"
					transferData={ { } }
					onDragStart={ onDragStart }
					onDragEnd={ onDragEnd }
				>
				{
					( { onDraggableStart, onDraggableEnd } ) => (
						<Dashicon
							icon="move"
							onDragStart={ onDraggableStart }
							onDragEnd={ onDraggableEnd }
							draggable
						/>
					)
				}
				</Draggable>
			</PanelBody>
		</Panel>
	</div>
);
```

I would love us to explore whether it could be refactored to:

```jsx
import { Dashicon, Draggable, Panel, PanelBody } from '@wordpress/components';

const MyDraggable = ( { onDragStart, onDragEnd } ) => (
	<Draggable
		as={ Panel }
		header="Draggable panel"
		transferData={ { } }
		onDragStart={ onDragStart }
		onDragEnd={ onDragEnd }
	>
		{ ( { onDraggableStart, onDraggableEnd } ) => (
			<PanelBody>
				<Dashicon
					icon="move"
					onDragStart={ onDraggableStart }
					onDragEnd={ onDraggableEnd }
					draggable
				/>
			</PanelBody>
		) }
	</Draggable>
);
```

/cc @nosolosw

Reply:
There is also a related issue with draggable elements as explained in:
https://github.com/WordPress/gutenberg/pull/18158

Reply:
> Why do you need to wrap it with a div which has the id which is referenced with elementId prop?

@gziolo I noticed that too. I found it a bit unintuitive/restrictive in my first couple attempts of getting `Draggable` to work.

The above `as` technique is cool! And if `as` is not defined, maybe it can fallback to a regular `<div>`, wrapping the `children` that has been passed in.